examples/hello/lib/hello/app.rb

```diff
@@ -0,0 +1,66 @@
+require "hello/actions"
+
+module Hello
+  class App
+    class Router
+      class NotFound < StandardError; end
+
+      attr_reader :routes
+
+      def initialize(&block)
+        @routes = {}
+        instance_eval(&block)
+      end
+
+      def resource(name, at:, &block)
+        @this_path = at
```

Comment:
This is so bad but it was the quickest way to get to a green passing test. Possible refactoring: init a `Router::Resource` object and instance eval the block in that context. That's where the `on` method should live and it could init a `Router::Route` object that holds the action name (and eventually the call args and responder). So instead of a nested hash, `Router` will hold a hash of `Resource`s and `Resource` will hold a hash of `Route`s.